### PR TITLE
Validate packages

### PR DESCRIPTION
**Pull request by freaktechnik**
Add a method to validate webapp packages.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,15 @@ fxos.validateManifest(manifestUrl).then(function(result) {
 
 `manifestUrl` is a hosted manifest.webapp URL. (i.e. http://brittanystoroz.github.io/its-five-o-clock-somewhere/manifest.webapp). The promise resolves with a validated manifest ID, that can then be used to publish your application to the marketplace.
 
+## Validating a packaged application
+```javascript
+fxos.validatePackage(packagePath).then(function(result) {
+  console.log('Result: ', result);
```
Comment (tofumatt) [inline README.md:42]: Reeeeeally nitpicking here, but if you have the comma separating `console.log` arguments, isn't the trailing space redundant?

**Comment by tofumatt** [comment]
Neat! :+1: (I'll defer to @brittanystoroz for final review)

**Comment by freaktechnik** [comment]
Oh, also, it should probably be possible to publish validated packages using the publish method, problem there is, that the method somehow has to know, if it's getting the ID of a validated manifest or a validated package.

File: README.md (hunk 1 above)
Comment (freaktechnik) [inline README.md:42]: The manifest validation example does the same, which I more or less just copied...

Comment (tofumatt) [inline README.md:42]: Fair enough, ignore my silly nitpicks :-)

**Comment by brittanydionigi** [comment]
@freaktechnik this looks great -- I'm confident it works, though when trying it out I'm getting an 'Error extracting manifest file' message. Likely there is just something wrong with the test zip I'm using. Do you happen to have a .zip handy that worked for you when trying this out?

This also means we need to be less specific with what we return.  If we resolve the promise with `body.id`, (as we're doing [here](https://github.com/mozilla/node-firefox-marketplace/pull/1/files#diff-168726dbe96b3ce427e7fedce31bb0bcR91) but you get an error (like the one I'm getting), it will resolve with `undefined`.

The `body` value being returned with my error is:

```
{ upload: [ 'Error extracting manifest from zip file.' ] }
```

**Comment by brittanydionigi** [comment]
It's bizarre that the request is considered successful if there was a problem extracting the manifest file.

**Comment by freaktechnik** [comment]
It works for me with this simple script: https://gist.github.com/freaktechnik/45037f0b2b4d77bf509f and the mines.zip from https://github.com/freaktechnik/mines.js/releases/tag/1.0.0

Also, when the API response for packaged apps comes back it's not guaranteed (well, only if it returns a 202, 201 means it processed), that the package is valid, as the validation is queued, which is not taken into account by this implementation. However you could validate your package using local validation tools before uploading.

Edit: I'm also wondering if handling the `valid: false` case as error would be a good idea. Edit2: ok, the manifest validation doesn't care about the validity and doesn't pass those errors as errors.

**Comment by freaktechnik** [comment]
It works for me with this simple script: https://gist.github.com/freaktechnik/45037f0b2b4d77bf509f and the mines.zip from https://github.com/freaktechnik/mines.js/releases/tag/1.0.0


Also, when the API response for packaged apps comes back it's not guaranteed, that the package is valid, as the validation is queued, which is not taken into account by this implementation. However you could validate your package using local validation tools before uploading.

**Comment by brittanydionigi** [comment]
@freaktechnik thanks for looking into this -- I just confirmed all of your findings as well. I was actually under the impression when I first wrote this that IDs were only returned if the manifest was valid. (Expected it to error otherwise.)

I'm going to merge this in and open new issues for documenting what's returned. Thanks for contributing!

**Pull request merged**